app/AppKernel.php

```diff
@@ -300,10 +302,13 @@ public function getProjectDir(): string
         return realpath(__DIR__ . '/..');
     }
 
-    protected function getModuleRepository(): ModuleRepository
+    protected function getModuleRepository(): CachedModuleRepository
     {
         if ($this->moduleRepository === null) {
-            $this->moduleRepository = new ModuleRepository(_PS_ROOT_DIR_, _PS_MODULE_DIR_);
+            $this->moduleRepository = new CachedModuleRepository(
+                new ModuleRepository(_PS_ROOT_DIR_, _PS_MODULE_DIR_),
+                new FilesystemAdapter('modules', 0, _PS_CACHE_DIR_)
```

Comment:
```suggestion
                new FilesystemAdapter('modules', 0, $this->getCacheDir())
```

Reply:
The tests are failing, probably because of the cache system now in place Since the tests modify the installed modules the values are not updated anymore

To overcome this issue you need to change the cache adapter in test environment:

```php
if ($this->environment === 'test') {
    // This cache is equivalent to no cache, in test environment we don't want the value to be cached
    $cache = new NullAdapter();
} else {
    $cache = new FilesystemAdapter(...);
}
```

That's where we see that relying on the `CacheInterface` is already relevant to adapt our use cases, and you probably need to implement something similar in the FO ContainerBuilder.

Let's hope that with this modification the tests will be green again

Reply:
@jolelievre I've made the changes, I hope I understood correctly.
We're waiting for the tests!